fetch/fetch.go

```diff
@@ -645,6 +640,9 @@ func (f *Fetch) RegisterPeerHashes(peer p2p.Peer, hashes []types.Hash32) {
 
 // AddPeersFromHash adds peers from one hash to others.
 func (f *Fetch) AddPeersFromHash(fromHash types.Hash32, toHashes []types.Hash32) {
+	f.hashToPeers.Mu.Lock()
+	defer f.hashToPeers.Mu.Unlock()
```

Comment:
if you manage to contain the locking inside cache.go, then there is no need to export the lock. just like what you suggested in DM, add AddPeersFromHash() to cache too and make this the wrapper.

Reply:
done, cheers